### PR TITLE
How to integrate safaricom lipa na mpesa in php

### LINKED ISSUE #4972: [Language] How to Intergrate  Safaricom Lipa Na Mpesa (STK Push)  in PHP
#### Proposed article introduction
Safaricom has introduced a new M-Pesa API dubbed ‘The Daraja API’ (Daraja in Swahili means bridge) to try and bridge this gap for developers.The Lipa na M-Pesa online service exposed by Daraja API is a very efficient way to implement merchant checkouts in e-commerce sites, mobile Apps, car parking management systems.

A common implementation  in PHP is, you have a checkout button in your  internet site which a user clicks to pay for the goods or services offered. This button when clicked triggers a post request to Daraja with the relevant payload. When Daraja receives the request it then triggers an STK push to the user’s phone prompting the user to enter his/her M-Pesa PIN to complete the transaction. When the user enters the PIN to complete the transaction, this request is sent to M-Pesa for processing, the user’s M-Pesa account is debited, and then you receive a webhook (to your servers webhook URL you provided) with the details of the transaction.

#### Key takeaways 
- Understand Daraja API
- How to intergrate Daraja API with PHP

#### Article quality
The article is going to be unique since am going to make it beginner friendly. Since it will allow the Customer to enter the phone number that he/she will used to pay irregardless of the phone number that he/she registered the account with. Also Most of the articles uses frameworks but mine will use pure PHP programming language.

#### References
- Safaricom Daraja Documentation.

### PR DESCRIPTION
**Pull request by HillaryKiprono**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4972

### CODE REVIEW

File: content/articles/how-to-integrate-safaricom-lipa-na-mpesa-in-php/index.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    </div>
+</body>
+</html>
+
+```
+### Step 5 - Writing our Mpesa Logic.
+Define a new file and name it `mpesa.php`. 
+
+#### Variables Declaration
+Inside the file declare and initialize the following variables using your credentials obtain from an app you created in  safaricom Daraja.
```
Comment (jumamiller) [inline content/articles/how-to-integrate-safaricom-lipa-na-mpesa-in-php/index.md:76]: Why are we declaring these variables?

File: content/articles/how-to-integrate-safaricom-lipa-na-mpesa-in-php/index.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            url_setopt($url, urlOPT_URL, $tokenUrl);
+            $credentials = base64_encode($consumer_key . ':' . $consumer_secret);
+            url_setopt($url, urlOPT_HTTPHEADER, array('Authorization: Basic ' . $credentials));
+            url_setopt($url, urlOPT_HEADER, false);
+            url_setopt($url, urlOPT_RETURNTRANSFER, 1);
+            url_setopt($url, urlOPT_SSL_VERIFYPEER, false);
+            $url_response = url_exec($url);
+
+            $token = json_decode($url_response)->access_token;
+```
```
Comment (jumamiller) [inline content/articles/how-to-integrate-safaricom-lipa-na-mpesa-in-php/index.md:107]: What's the use of this auth token?

Comment (jumamiller) [reply content/articles/how-to-integrate-safaricom-lipa-na-mpesa-in-php/index.md:107]: This code doesn't work!

File: content/articles/how-to-integrate-safaricom-lipa-na-mpesa-in-php/index.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                'transaction' => $transaction,
+                'Amount' => $amount,
+                'PartyA' => $phone,
+                'PartyB' => $BusinessShortCode,
+                'PhoneNumber' => $phone,
+                'CallBackURL' => $CallBackURL,
+                'account' => 'Test',
+                'TransactionDesc' => 'Test',
+            ];
+
```
Comment (jumamiller) [inline content/articles/how-to-integrate-safaricom-lipa-na-mpesa-in-php/index.md:131]: The whole code cannot work!

File: content/articles/how-to-integrate-safaricom-lipa-na-mpesa-in-php/index.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            </form>
+```
+
+### Demo
+![Demo1](/engineering-education/how-to-integrate-safaricom-lipa-na-mpesa-in-php/demo1.png)
+
+![Demo2](/engineering-education/how-to-integrate-safaricom-lipa-na-mpesa-in-php/demo2.png)
+
+![Demo3](/engineering-education/how-to-integrate-safaricom-lipa-na-mpesa-in-php/demo3.png)
+
```
Comment (jumamiller) [inline content/articles/how-to-integrate-safaricom-lipa-na-mpesa-in-php/index.md:167]: Attach these outputs on each step

**Review by jumamiller — CHANGES_REQUESTED** [review]
Explain your code snippets! The current code doesn't work too!

**Comment by jumamiller** [comment]
Any update @HillaryKiprono ?

**Comment by Kayere** [comment]
Hello @HillaryKiprono.

Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[How to integrate safaricom lipa na mpesa in php #5246 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7660573/How.to.integrate.safaricom.lipa.na.mpesa.in.php.5246.Copyleaks.Report.pdf)

**Comment by Kayere** [comment]
@HillaryKiprono - The article seems to be a bit short and of low substance. Please add more content to the article to give it more value.

**Pull request closed without merging**